test/oru/index.js

```diff
@@ -0,0 +1,39 @@
+import test from 'ava'
+import {decode} from '../../src/lib/parser'
+import r01Mapping from './config/r01.json'
+
+test(`ORU - Notification of new observation result`, t => {
+  const r01 = `MSH|^~\&|LCS|LCA|LIS|TEST9999|199807311532||ORU^R01|3629|P|2.2\rPID|2|2161348462|20809880170|1614614|20809880170^TESTPAT||19760924|M|||^^^^\r00000-0000|||||||86427531^^^03|SSN# HERE\rORC|NW|8642753100012^LIS|20809880170^LCS||||||19980727000000|||HAVILAND\rOBR|1|8642753100012^LIS|20809880170^LCS|008342^UPPER RESPIRATORY\rCULTURE^L|||19980727175800||||||SS#634748641 CH14885 SRC:THROA\rSRC:PENI|19980727000000||||||20809880170||19980730041800||BN|F\rOBX|1|ST|008342^UPPER RESPIRATORY CULTURE^L||FINALREPORT|||||N|F||| 19980729160500|BN\rORC|NW|8642753100012^LIS|20809880170^LCS||||||19980727000000|||HAVILAND\rOBR|2|8642753100012^LIS|20809880170^LCS|997602^.^L|||19980727175800||||G|||\r19980727000000||||||20809880170||19980730041800|||F|997602|||008342\rOBX|2|CE|997231^RESULT 1^L||M415|||||N|F|||19980729160500|BN\rNTE|1|L|MORAXELLA (BRANHAMELLA) CATARRHALIS\rNTE|2|L| HEAVY GROWTH\rNTE|3|L| BETA LACTAMASE POSITIVE\rOBX|3|CE|997232^RESULT 2^L||MR105|||||N|F|||19980729160500|BN\rNTE|1|L|ROUTINE RESPIRATORY FLORA`
+  const obj = decode(r01, r01Mapping)
+  t.is(obj.msh.message_datetime, '199807311532')
+  t.is(obj.msh.message_type, 'ORU')
+  t.is(obj.msh.message_type_ref, 'R01')
+  t.is(obj.msh.message_control_id, '3629')
+  t.is(obj.pid.id, '20809880170')
+  t.is(obj.pid.origin, '')
+  t.is(obj.pid.first_name, 'TESTPAT')
+  t.is(obj.pid.last_name, '20809880170')
+  t.is(obj.pid.birthdate, '19760924')
+  t.is(obj.pid.gender, 'M')
+  t.is(obj.pid.street_name, '')
+  t.is(obj.pid.city, '')
+  t.is(obj.pid.zip_code, '')
+  t.is(obj.pid.phone, '')
+  t.is(obj.pid.email, '')
+  t.is(obj.nte.length, 4)
+  t.is(obj.nte[0].id, '1')
+  t.is(obj.nte[0].source, 'L')
+  t.is(obj.nte[0].comment, 'MORAXELLA (BRANHAMELLA) CATARRHALIS')
+  t.is(obj.nte[1].id, '2')
+  t.is(obj.nte[1].source, 'L')
+  t.is(obj.nte[1].comment, ' HEAVY GROWTH')
+  t.is(obj.nte[2].id, '3')
+  t.is(obj.nte[2].source, 'L')
+  t.is(obj.nte[2].comment, ' BETA LACTAMASE POSITIVE')
+  t.is(obj.nte[3].id, '1')
+  t.is(obj.nte[3].source, 'L')
+  t.is(obj.nte[3].comment, 'ROUTINE RESPIRATORY FLORA')
+  t.is(obj.orc.length, 2)
+  t.is(obj.obr.length, 2)
+  t.is(obj.obx.length, 3)
```

Comment:
Didn't add all expectations here for all these segments since I think that functionality is already covered in the NTE segments.

Reply:
Don't worry, it's enough. It's a good job!